1.Exploratory_Data_Analysis/scripts/5.anova.py

```diff
@@ -0,0 +1,385 @@
+
+# create a venn diagram of the features that are significant in all conditions
+from matplotlib_venn import venn2, venn3
+
+warnings.filterwarnings("ignore")
+from pycytominer.cyto_utils import infer_cp_features
+from scipy.stats import f_oneway
+from statsmodels.formula.api import ols
+from statsmodels.stats.multicomp import MultiComparison, pairwise_tukeyhsd
+
+# In[2]:
+
+
+# Parameters
+cell_type = "PBMC"
+treatment1 = "DMSO_0.100_DMSO_0.025"
+treatment2 = "LPS_0.010_DMSO_0.025"
+treatment3 = "H202_100_DMSO_0.025"
+
+
+# In[3]:
+
+
+# Import Data
+# set data file path under pathlib path for multi-system use
+file_path = pathlib.Path(f"../data/{cell_type}_preprocessed_sc_norm.parquet")
+df = pq.read_table(file_path).to_pandas()
+df_metadata = df.filter(regex="Metadata")
+df_data = df.drop(df_metadata.columns, axis=1)
+cp_features = infer_cp_features(df)
+
+
+# In[4]:
+
+
+trt1 = df[
+    df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"].str.contains(f"{treatment1}")
+]
+trt2 = df[
+    df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"].str.contains(f"{treatment2}")
+]
+trt3 = df[
+    df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"].str.contains(f"{treatment3}")
+]
+
+
+# In[5]:
+
+
+# df_LPS_100 = df[df['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].str.contains('LPS_100.000_DMSO_0.025')]
+# df_LPS_100['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].unique()
+# df_LPS_10 = df[df['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].str.contains('LPS_10.000_DMSO_0.025')]
+# df_LPS_10['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].unique()
+# df_flag = df[df['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].str.contains('Flagellin_1.000_DMSO_0.025')]
+# df_flag['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].unique()
+# df_DMSO = df[df['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].str.contains('DMSO_0.100_DMSO_0.025')]
+# df_DMSO['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].unique()
+# df_thapsi10 = df[df['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].str.contains('Thapsigargin_10.000_DMSO_0.025')]
+# df_thapsi10['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].unique()
+# df_thapsi_DMSO = df[df['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].str.contains('Thapsigargin_10.000_DMSO_0.025','DMSO_0.100_DMSO_0.025')]
+# df_thapsi_DMSO['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].unique()
+# df_thapsi1 = df[df['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].str.contains('Thapsigargin_1.000_DMSO_0.025')]
+# df_thapsi1['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].unique()
+
+
+# ### Set up DF
+
+# In[6]:
+
+
+combined_df = pd.concat([trt1, trt2, trt3], axis=0)
+print(len(trt1), len(trt2), len(trt3), len(combined_df))
+combined_df.head(3)
+
+
+# ## Anova + Post Hoc testing (tukeyHSD)
+
+# In[7]:
+
+
+# anova for each feature in the dataframe with posthoc tukey test to determine which groups are different from each other
+lst = []
+for i in cp_features:
+    formula = f"{i} ~ C(oneb_Metadata_Treatment_Dose_Inhibitor_Dose) + C(Metadata_number_of_singlecells)"
```

Comment:
What is this formula? I haven't seen this before and curious what this is doing.

Reply:
@jenna-tomkinson The formula is the linear model I am using to fit the data to. This allows for variates and co-variates to be used to calculate <ins>**A**</ins> <ins>**N**</ins>umber <ins>**O**</ins>f  <ins>**VA**</ins>riances (ANOVA)